core-plugins/src/main/java/io/cdap/plugin/batch/aggregator/function/MaxSelection.java

```diff
@@ -46,4 +48,14 @@ protected int compareFloat(float val1, float val2) {
   protected int compareDouble(double val1, double val2) {
     return Double.compare(val1, val2);
   }
+
+  @Override
+  protected int compareBytes(ByteBuffer val1, ByteBuffer val2) {
+    return val1.compareTo(val2);
+  }
+
+  @Override
+  protected int compareString(String val1, String val2) {
+    return String.CASE_INSENSITIVE_ORDER.compare(val1, val2);
```

Comment:
Why case insensitive? Is it confirmed by product?

Reply:
What is the date/time format used? Are we sure string comparison order will follow date value order? Could you add a test

Reply:
uh shouldn't it ideally be case insensitive?

Reply:
yeah I'll add a test

Reply:
Well, it depends on product. As an engineer I would not expect my max function to be case-insensitive by default

Reply:
And if we do only dates here, I'd not make it case insensitive as it just adds more processing

Reply:
Ah okay I get it. Thanks!